### PR TITLE
net/sslsplit: Added a package for SSLsplit (https://www.roe.ch/SSLsplit)

### PR DESCRIPTION
**Pull request by phymod0**
Maintainer: me
Compile tested: ar71xx, TP-Link (TL-MR3020), LEDE 17.01
Run tested: ar71xx, TP-Link (TL-MR3020), LEDE 17.01

Description:
SSLsplit package (see [the official page](https://www.roe.ch/SSLsplit)) for MITM attacks against SSL/TLS connections.

Signed-off-by: Awais Chishti <chishtiawais@hotmail.com>

### CODE REVIEW

File: net/sslsplit/Makefile [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (C) 2016-2018 OpenWrt.org
```
Comment (diizzyy) [inline net/sslsplit/Makefile]: Don't assign copyright to OpenWrt.org unless you're affiliated, either omit or use your own.

File: net/sslsplit/Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=sslsplit
+PKG_VERSION:=0.5.2
+PKG_RELEASE:=1
+PKG_LICENSE:=BSD-2-Clause
+PKG_LICENSE_FILES:=LICENSE LICENSE.contrib LICENSE.third
+
+PKG_SOURCE:=sslsplit.tar.gz
```
Comment (diizzyy) [inline net/sslsplit/Makefile]: Use release tarball instead

**Comment by diizzyy** [comment]
I have a pull request open for 3proxy (without SSL support for now) which covers SSLsplit's functionality to some extent and has less dependencies which may be of interest.
https://3proxy.ru/plugins/SSLPlugin/

File: net/sslsplit/Makefile (hunk 2 above)
Comment (phymod0) [reply net/sslsplit/Makefile]: Do you mean I should rename sslsplit.tar.gz to sslsplit-0.5.2.tar.gz?

Comment (diizzyy) [reply net/sslsplit/Makefile]: https://mirror.roe.ch/rel/sslsplit/sslsplit-0.5.2.tar.bz2

**Comment by phymod0** [comment]
@diizzyy I made the changes you suggested, thanks for the feedback.
Also I thought I should ask if it is fine to submit this PR considering that I'm not involved in the SSLsplit project. And does "Maintainer" (as mentioned in the first comment) refer to the person maintaining just the files for the package (me) or the maintainer of SSLsplit?

**Comment by diizzyy** [comment]
You're welcome
Maintainer refers to package maintainer (in this repo) :-)

File: net/sslsplit/Makefile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+PKG_INSTALL:=1
+
+PKG_MAINTAINER:=Awais Chishti <chishtiawais511@gmail.com>
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/sslsplit
+  SECTION:=net
+  CATEGORY:=Network
+  DEPENDS:=+libevent2 +libopenssl +libevent2-openssl +libevent2-pthreads +USE_MUSL:musl-fts
```
Comment (neheb) [inline net/sslsplit/Makefile]: Is libevent2-openssl needed?

**Comment by neheb** [comment]
@phymod0 please rebase against current master to build against CircleCI.

**Comment by phymod0** [comment]
@neheb The CircleCI checks pass after rebasing. Anything else I should do?

File: net/sslsplit/Makefile [hunk 4]
```diff
@@ -0,0 +1,46 @@
+
```
Comment (diizzyy) [inline net/sslsplit/Makefile]: Remove

File: net/sslsplit/Makefile [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=sslsplit
+PKG_VERSION:=0.5.2
```
Comment (diizzyy) [inline net/sslsplit/Makefile]: 0.5.4 is out

File: net/sslsplit/Makefile [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  DEPENDS:=+libevent2 +libopenssl +libevent2-openssl +libevent2-pthreads +USE_MUSL:musl-fts
+  TITLE:=Transparent SSL/TLS interception
+  URL:=https://www.roe.ch/SSLsplit
+endef
+
+define Package/sslsplit/description
+  Transparent SSL/TLS interception
+endef
+
+MAKE_FLAGS += prefix=/usr \
```
Comment (diizzyy) [inline net/sslsplit/Makefile]: Needed?

File: net/sslsplit/Makefile (hunk 5 above)
Comment (phymod0) [reply net/sslsplit/Makefile]: 0.5.4 has some dependency changes. I'm working on it.

File: net/sslsplit/Makefile (hunk 6 above)
Comment (phymod0) [reply net/sslsplit/Makefile]: Can you specify what you're referring to exactly?

Comment (diizzyy) [reply net/sslsplit/Makefile]: nm, the Makefile hardsets /usr/local

**Comment by neheb** [comment]
GNUmakefile:317: *** dependency 'libpcap' not found; install it or point LIBPCAP_BASE to base path.  Stop.

**Comment by diizzyy** [comment]
Closing this due to timeout

**Pull request closed without merging**